### PR TITLE
Homework es2015

### PR DESCRIPTION
**Pull request by Pristavka**

### CODE REVIEW

File: css/style.css [hunk 1]
```diff
@@ -0,0 +1,125 @@
+html, body{
```
Comment (NikitaSedyx) [inline css/style.css]: it will be great to add empty lines between selectors and before `{`

File: css/style.css [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  }
+  footer{
+    display: flex;
+    justify-content: space-evenly;
+    align-items: center;
+    flex-wrap: wrap;
+    min-height: 70px;
+    background: #999;
+    color:#fff;
+  }
```
Comment (NikitaSedyx) [inline css/style.css]: eof

File: js/common.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  }
+  getMarcaNews = () => {
+    getNews('https://newsapi.org/v2/top-headlines?sources=marca&apiKey=eefc0a2a311a42dc8936f6dceba17753')
+      .then((allnews) => renderSportNews(allnews));
+  }
+
+  document.querySelector('.espn').addEventListener('click', getESPNNews);
+  document.querySelector('.nhl').addEventListener('click', getNHLNews);
+  document.querySelector('.marca').addEventListener('click', getMarcaNews);
+})()
```
Comment (NikitaSedyx) [inline js/common.js]: same

File: js/common.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
+; (() => {
+  setCurrentDate = () => {
```
Comment (NikitaSedyx) [inline js/common.js]: global variable? o_O

File: js/common.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+  // Common getNews method
+  getNews = (url) => {
+    return fetch(url)
+      .then((resp) => resp.json())
+  }
+
+  // Functionality for fetching and rendering Latest News
+  getLatestNews = () => {
+    getNews('https://newsapi.org/v2/top-headlines?sources=bbc-news&apiKey=eefc0a2a311a42dc8936f6dceba17753')
```
Comment (NikitaSedyx) [inline js/common.js]: it will be great to make api_key as a constant
FYI api keys are secure info

File: js/common.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
+; (() => {
+  setCurrentDate = () => {
+    let date = new Date();
```
Comment (NikitaSedyx) [inline js/common.js]: looks like it's `const`

File: js/common.js [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+      })
+  }
+  getLatestNews();
+
+  // Functionality for fetching and rendering Sport News
+  const listOfsportNews = document.querySelector('.listOfsportNews');
+  renderSportNews = (allnews) => {
+    listOfsportNews.innerHTML = '';
+    allnews.articles.forEach((art) => {
+      let div = document.createElement('div');
```
Comment (NikitaSedyx) [inline js/common.js]: same

File: js/common.js [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+        document.querySelector('.latest__itemNews').innerHTML = news;
+      })
+  }
+  getLatestNews();
+
+  // Functionality for fetching and rendering Sport News
+  const listOfsportNews = document.querySelector('.listOfsportNews');
+  renderSportNews = (allnews) => {
+    listOfsportNews.innerHTML = '';
+    allnews.articles.forEach((art) => {
```
Comment (NikitaSedyx) [inline js/common.js]: brackets for 1 argument can be omitted

File: js/view.js [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    const source = `
+      <div class="sourceItem">
+        <p><img src="https://icons.better-idea.org/icon?url=${src.url}&size=70..120..200" alt="${src.name}" id="${src.id}"</p>
+        <p class="sport__title">${src.name}</p>
+      </div>
+    `
+    div.innerHTML = source;
+    container.appendChild(div);
+    document.querySelector(`#${src.id}`)
+      .addEventListener('click', () => getSportNews(`https://newsapi.org/v2/top-headlines?sources=${src.id}&apiKey=${api_key}`))
```
Comment (NikitaSedyx) [inline js/view.js]: it's not cool to add listener to each item, why not to handle parent events?

File: js/api.js [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+};
+
+const getSportNews = url => {
+  getNews(url)
+    .then(allnews => renderSportNews(allnews));
+};
+
+const getSportSources = url => {
+  fetch(url)
+    .then(src => src.json())
```
Comment (NikitaSedyx) [inline js/api.js]: looks like it's copy paste

Comment (NikitaSedyx) [reply js/api.js]: FYI `fetch` is resolved when response has error status

File: js/controller.js [hunk 11]
```diff
@@ -0,0 +1,14 @@
+const setCurrentDate = () => {
+  const date = new Date();
+  const options = {
+    year: 'numeric',
+    month: 'long',
+    day: 'numeric',
+    weekday: 'long',
+  };
+  document.querySelector('.header__date').innerHTML = date.toLocaleString("en-US", options)
```
Comment (NikitaSedyx) [inline js/controller.js]: please chose one code style one or two quotes

File: js/view.js [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+        <img src="${art.urlToImage}" alt="${art.title}">
+        <p class="latest__title">${art.title}</p>
+        <p class="latest__url"><a href="${art.url}" target="_blank">Open original source</a></p>
+      `
+    div.innerHTML = news;
+    listOfsportNews.appendChild(div);
+  })
+};
+
+const renderSportSources = sources => {
```
Comment (NikitaSedyx) [inline js/view.js]: Destructuring to have sources instead of `sources.sources`

File: js/view.js [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+    <p class="latest__url"><a href="${latestNews.articles[0].url}" target="_blank">Open original source</a></p>
+    <p class="latest__published"><b>Date of publication:</b> ${latestNews.articles[0].publishedAt}</p>
+  `
+  document.querySelector('.latest__itemNews').innerHTML = news;
+};
+
+const listOfsportNews = document.querySelector('.listOfsportNews');
+const renderSportNews = allnews => {
+  listOfsportNews.innerHTML = '';
+  allnews.articles.forEach(art => {
```
Comment (NikitaSedyx) [inline js/view.js]: same with sources.sources

File: js/view.js [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+  })
+};
+
+const renderSportSources = sources => {
+  const container = document.querySelector('.sportSourcesContainer');
+  sources.sources.forEach(src => {
+    const div = document.createElement('div');
+    const source = `
+      <div class="sourceItem">
+        <p><img src="https://icons.better-idea.org/icon?url=${src.url}&size=70..120..200" alt="${src.name}" id="${src.id}"</p>
```
Comment (NikitaSedyx) [inline js/view.js]: long line

File: js/view.js (hunk 9 above)
Comment (NikitaSedyx) [inline js/view.js]: looks like I've seen this `https://newsapi.org/v2/top-headlines?sources=` above

File: js/view.js [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+const listOfsportNews = document.querySelector('.listOfsportNews');
+const renderSportNews = allnews => {
+  listOfsportNews.innerHTML = '';
+  allnews.articles.forEach(art => {
+    const div = document.createElement('div');
+    div.className = 'sport__item';
+    const news = `
+        <img src="${art.urlToImage}" alt="${art.title}">
+        <p class="latest__title">${art.title}</p>
+        <p class="latest__url"><a href="${art.url}" target="_blank">Open original source</a></p>
```
Comment (NikitaSedyx) [inline js/view.js]: please read about `rel=nooponer` for links with `target=_blank`

File: config/config.js [hunk 16]
```diff
@@ -0,0 +1,11 @@
+const api_key = 'eefc0a2a311a42dc8936f6dceba17753';
+const resources = {
+  bbc: `https://newsapi.org/v2/top-headlines?sources=bbc-news&apiKey=${api_key}`,
+  sources: `https://newsapi.org/v2/sources?category=sport&apiKey=${api_key}`
```
Comment (NikitaSedyx) [inline config/config.js]: `https://newsapi.org/v2` looks the same

File: js/api.js [hunk 17]
```diff
@@ -0,0 +1,4 @@
+const getData = url => {
```
Comment (NikitaSedyx) [inline js/api.js]:
```
const getData = url => fetch(url).then(resp => resp.json())
```

File: js/controller.js [hunk 18]
```diff
@@ -0,0 +1,25 @@
+const getLatestNews = (url) => {
+  getData(url)
+    .then(latestNews => {
```
Comment (NikitaSedyx) [inline js/controller.js]: `.then(renderLatestNews)`

File: js/controller.js [hunk 19]
```diff
@@ -0,0 +1,25 @@
...
+      renderLatestNews(latestNews);
+    })
+}
+
+const getSportNews = (url) => {
+  getData(url)
+    .then(sportNews => {
```
Comment (NikitaSedyx) [inline js/controller.js]: same

File: js/controller.js [hunk 20]
```diff
@@ -0,0 +1,25 @@
...
+      renderSportNews(sportNews);
+    })
+}
+
+const getSportSources = (url) => {
+  getData(url)
+    .then(src => {
+      renderSportSources(src);
```
Comment (NikitaSedyx) [inline js/controller.js]: here too

File: js/view.js [hunk 21]
```diff
@@ -0,0 +1,73 @@
+const setCurrentDate = () => {
+  const date = new Date();
+  document.querySelector('.header__date').innerHTML = date.toLocaleString("en-US", timeOptions)
```
Comment (NikitaSedyx) [inline js/view.js]: it could be `(new Date()).toL...`

File: js/view.js (hunk 12 above)
Comment (Pristavka) [reply js/view.js]: fixed

File: js/view.js (hunk 14 above)
Comment (Pristavka) [reply js/view.js]: fixed

File: js/view.js (hunk 15 above)
Comment (Pristavka) [reply js/view.js]: thx useful info, fixed

File: config/config.js (hunk 16 above)
Comment (Pristavka) [reply config/config.js]: fixed

File: js/api.js (hunk 17 above)
Comment (Pristavka) [reply js/api.js]: improved

File: js/controller.js (hunk 18 above)
Comment (Pristavka) [reply js/controller.js]: improved

File: js/controller.js (hunk 19 above)
Comment (Pristavka) [reply js/controller.js]: improved

File: js/controller.js (hunk 20 above)
Comment (Pristavka) [reply js/controller.js]: improved

File: js/view.js (hunk 21 above)
Comment (Pristavka) [reply js/view.js]: improved

**Pull request merged**